### ISSUE TITLE
add url prefix to hosting multiple instances 

### ISSUE THREAD

**Issue by yywill** [elkozmon/zoonavigator#27]
Hi,

I have only one host to server multiple web project, such that to access http://myhost/app_prefix/ to reverse proxy to different web ui.

I found this: https://www.elkozmon.com/zoonavigator/get-started/usage/ the environment section that does not have prefix or path varible. 

so, I cannot access let nginx do the proxy:
such as:

```
 location ~ ^/zoonavigator/(.*)$ {
                proxy_pass              http://localhost:8000;
                rewrite /zoonavigator/(.*) /$1  break;
                proxy_set_header        Host $host;
                proxy_set_header        Referer "";
                proxy_set_header        X-Real-IP $remote_addr;
                proxy_set_header        X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_http_version      1.1;
                proxy_connect_timeout   150;
                proxy_send_timeout      100;
                proxy_read_timeout      100;
                proxy_buffers           16 64k;
                proxy_busy_buffers_size 64k;
                client_max_body_size    256M;
                client_body_buffer_size 128k;
 }
```

and the screen shot:

![screen shot 2018-09-28 at 3 16 45 pm](https://user-images.githubusercontent.com/1792088/46193686-9a0a9a00-c331-11e8-834f-ebc301a7a641.png)

those resources are all point to root of my domain. 

Can add one more env variable?

**Comment by elkozmon** [comment]
Sorry for getting back to you so late. I am currently working on this, however, it's a bit tricky. In Angular you have to provide the "prefix" at the build time, which happens during Docker image build, so it cannot be simply added as an environment variable to the final image.

**Comment by elkozmon** [comment]
Hi, I updated the documentation [with instructions](https://www.elkozmon.com/zoonavigator/how-tos/custom-url/) on how to deploy ZooNavigator with custom URL. In short, you will have to build your own Docker image where you'll provide the URL prefix.

**Comment by mujina93** [comment]
Link is broken

**Comment by elkozmon** [comment]
@mujina93 Thanks for the notice, I've updated the link. Also, I realized that the instructions were wrong (submodule init command) so I've updated that as well.

**Comment by mujina93** [comment]
@elkozmon Thanks for the quick answer!

A couple more things:

- The instructions for the submodule(s) fail as the default is to get the via ssh. I was able to get past `git submodule update --init --recursive` by changing the urls in my `.gitmodules` to the http ones of your 2 submodules, followed by `git submodule sync`. Maybe it's worth adding these prior to the first instruction.

- I was able to build the custom docker image. Then I try to run it with the same command I was using before:

```
docker run -d --network host -e HTTP_PORT=9000 --name rev-zoonavigator --restart unless-stopped my-zoonavigator
```

it starts, I'm able to reach it at `<host>:9000/zoonavigator` (but not at `<host>:9000/zoonavigator/` with a `/` at the end), but then it's stuck loading the page. I only see the loading logo.

If I check the network in the browser console I see that it tries to make a request to `http://93.90.193.89:9000/api/config`.

Am I getting something wrong when trying to start it up or the basepath is missing somewhere (when `/api/config` is called)?

**Comment by elkozmon** [comment]
@mujina93 Good point, thanks. I've already replaced those URLs by https. I believe the instructions are now correct.

As for the problem with base path, I'm not sure right now but it definitely sounds like a bug to me.
It would be best to address this with a [new issue on Github](https://github.com/elkozmon/zoonavigator/issues/51).
